### PR TITLE
tests: Bluetooth: Mesh: Use bsim settings backend

### PR DESCRIPTION
**Pull request by alxelax**
PR adds using nrf52_bsim platform NVMC implementation as well as flash for settings and NVS instead of proprietary custom settings backend.

### CODE REVIEW

**Review by Andrewpini — DISMISSED** [review]
LGTM 👍 One minor comment.

**Review by Andrewpini — DISMISSED** [review]

**Review by PavelVPV — CHANGES_REQUESTED** [review]

File: tests/bsim/bluetooth/mesh/src/test_dfu.c [hunk 1]
```diff
@@ -583,13 +580,11 @@ static void test_dist_dfu_slot_create(void)
...
 		    "CONFIG_BT_MESH_DFU_SLOT_CNT must be at least 3");
 
-	bt_mesh_test_host_files_remove();
-
 	bt_mesh_test_cfg_set(NULL, WAIT_TIME);
 	bt_mesh_device_setup(&prov, &dist_comp);
 	dist_prov_and_conf(DIST_ADDR);
 
-	for (i = 0; i < CONFIG_BT_MESH_DFU_SLOT_CNT; i++) {
+	for (i = CONFIG_BT_MESH_DFU_SLOT_CNT - 1; i >= 0; i--) {
```
[the remark this answers is not in the document]
Comment (alxelax) [reply tests/bsim/bluetooth/mesh/src/test_dfu.c:587]: Nop. Neither commit with fix first nor bsim backend first. They will not pass bsim test. This is workaround DFU slot bug when slot sequence depends on settings backend. It should be fixed here: https://github.com/zephyrproject-rtos/zephyr/pull/61389

File: tests/bsim/bluetooth/mesh/src/mesh_test.c [hunk 2]
```diff
@@ -256,7 +255,10 @@ void bt_mesh_device_setup(const struct bt_mesh_prov *prov, const struct bt_mesh_
 
 	if (IS_ENABLED(CONFIG_BT_SETTINGS)) {
 		LOG_INF("Loading stored settings");
-		settings_load();
+		if (IS_ENABLED(CONFIG_BT_MESH_USES_MBEDTLS_PSA)) {
+			settings_load_subtree("itsemul");
+		}
+		settings_load_subtree("bt");
```
[the remark this answers is not in the document]
Comment (alxelax) [reply tests/bsim/bluetooth/mesh/src/mesh_test.c:261]: Keys storage area does not belong to Bluetooth. Restoring sequence within the same area is not guaranteed, but keys should be ready at the time when Bluetooth mesh is restored.

File: tests/bsim/bluetooth/mesh/src/psa_its_emul.c [hunk 3]
```diff
@@ -95,7 +95,7 @@ static int itsemul_set(const char *name, size_t len_rd, settings_read_cb read_cb
 	return 0;
 }
 
-BT_MESH_SETTINGS_DEFINE(psa_its_emu, "itsemul", itsemul_set);
+SETTINGS_STATIC_HANDLER_DEFINE(psa_its_emu, "itsemul", NULL, itsemul_set, NULL, NULL);
```
[the remark this answers is not in the document]
Comment (alxelax) [reply tests/bsim/bluetooth/mesh/src/psa_its_emul.c:98]: Keys storage area does not belong to Bluetooth. Restoring sequence within the same area is not guaranteed, but keys should be ready at the time when Bluetooth mesh is restored.

File: tests/bsim/bluetooth/mesh/src/test_dfu.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (PavelVPV) [reply tests/bsim/bluetooth/mesh/src/test_dfu.c:587]: I think it should be fine to move this change into separate commit which can be reverted later.

File: tests/bsim/bluetooth/mesh/src/mesh_test.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (PavelVPV) [reply tests/bsim/bluetooth/mesh/src/mesh_test.c:261]: Could you please move this change into a separate commit? This should work independently of the settings backend.

File: tests/bsim/bluetooth/mesh/_mesh_test.sh [hunk 4]
```diff
@@ -26,20 +26,32 @@ function RunTest(){
...
+    fi
+
     shift 1
   done
 
   test_options=$@
 
-  for testid in ${testids[@]} ; do
+  for testid in ${testid_in_order[@]}; do
+    echo "Test id: " $testid
```
Comment (PavelVPV) [inline tests/bsim/bluetooth/mesh/_mesh_test.sh:53]: This shouldn't be needed?

**Review by PavelVPV — APPROVED** [review]

**Review by Andrewpini — APPROVED** [review]

**Review by Balaklaka — APPROVED** [review]

**Pull request merged**